AppBar documentation (#1749)
telerik/blazor-docs: components/appbar/separators.md

---
title: Separators
page_title: AppBar - Separators
description: Separate items in the Telerik AppBar for Blazor
slug: appbar-separators
tags: telerik,blazor,appbar,navbar,separator,spacer
published: True
position: 5
---

# Separators

> You can visually separate the items in the Telerik AppBar for Blazor. Depending on the needs of your application you can use of the following, or a combination of them.

Reviewer: The introductory sentence should be self-sufficient to some extent, so that the reader finds out what the article is about, and if it's relevant to the reader's needs.

```suggestion
You can visually separate the items in the Telerik AppBar for Blazor. The component supports lines and empty space between its sections.
```

## AppBar Separator

To separate two items with a solid line, add the `<AppBarSeparator>` tag between them.

### AppBar Separator Parameters

> The nested `AppBarSeparator` tag exposes parameters:

Reviewer 2: ```suggestion
The nested `AppBarSeparator` tag exposes the following parameters:
```

| Parameter | Type and Default Value | Description |
| ----------- | ----------- | ----------- |

> | `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the AppBar separator. You could use that class to cascade styles. |

Reviewer 2: ```suggestion
| `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the AppBar separator. Use that class to cascade styles. |
```

| `Visible` | `bool` <br /> (`true`) | Specifies if the separator will be visible in the AppBar. |

## AppBar Spacer

To separate two items with a solid line, add the `<AppBarSeparator>` tag between them.

Use the `<AppBarSpacer>` to define empty space in the AppBar which separates the items.

### AppBar Spacer Parameters

> The nested `AppBarSpacer` tag exposes parameters:

Reviewer 2: ```suggestion
The nested `AppBarSpacer` tag exposes the following parameters:
```

| Parameter | Type and Default Value | Description |
| ----------- | ----------- | ----------- |

> | `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the AppBar spacer. You could use that class to cascade styles. |

Reviewer 2: ```suggestion
| `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the AppBar spacer. Use that class to cascade styles. |
```

> | `Size` | `string` | The width of the spacer. If you add an `AppBarSpacer` tag without Size the spacer will take all available space between the [Sections]({%slug appbar-sections%}). |

Reviewer 2: ```suggestion
| `Size` | `string` | The width of the spacer. If you add an `AppBarSpacer` tag without Size, the spacer will take all available space between the [Sections]({%slug appbar-sections%}). |
```

> | `Size` | `string` | The width of the spacer. If you add an `AppBarSpacer` tag without Size the spacer will take all available space between the [Sections]({%slug appbar-sections%}). |

Reviewer: A more precise explanation would be that all spacers without a width will take up the same amount of the remaining available space.

| `Visible` | `bool` <br /> `true` | Specifies if the spacer will be visible in the AppBar. |

> >caption The AppBar separators

Reviewer: ```suggestion
>caption Using AppBar separators
```

````CSHTML
@* The AppBar content dividers with some of their parameters and values *@

<style>
    .k-appbar .k-appbar-separator.thick-separator {
        border-width: 0 0 0 20px;
    }
</style>

<TelerikAppBar>
    <AppBarSection>
        <span>Our Logo</span>
    </AppBarSection>

    <AppBarSpacer Size="25%"></AppBarSpacer>

    <AppBarSection>
        <span>Our Products</span>
    </AppBarSection>

    <AppBarSpacer Size="50px"></AppBarSpacer>

    <AppBarSection>
        <span>Our Mission</span>
    </AppBarSection>

    <AppBarSpacer></AppBarSpacer>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.User"></TelerikSvgIcon>
    </AppBarSection>

    <AppBarSeparator Class="thick-separator"></AppBarSeparator>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.Logout"></TelerikSvgIcon>
    </AppBarSection>
</TelerikAppBar>
````

## See Also

>   * [Live Demo: AppBar Overview](https://demos.telerik.com/blazor-ui/appbar/overview)

Reviewer: Does the AppBar have only one live demo, so that we always link to it?
